extend/contribute/api-docs/overview.md

```diff
@@ -0,0 +1,113 @@
+
+## High-level process
+
+All Elastic API docs follows this general pattern:
+
+<!-- 
+flowchart TD
+    A[Source files] --> B[OpenAPI documents]
+    B --> C[Published documentation]
+    
+    style A fill:#fff2cc
+    style B fill:#e1d5e7
+    style C fill:#dae8fc
+-->
+
+:::{image} images/api-docs-general-pipeline.png
+:alt: High-level API docs workflow pipeline diagram, showing the flow from source files to OpenAPI documents and published documentation.
+:width: 490px
+:::
+
+1. **Source files** can be:
+    - TypeScript definitions with JSDoc comments (Elasticsearch)
+    - Code annotations in application source (some Kibana teams)  
+    - Manual YAML/JSON specifications (Logstash)
+    - Generated specifications from service code (Cloud teams)
+2. **OpenAPI documents** are either generated from source files or edited manually. This is the common intermediate format regardless of origin.
+3. **Published documentation** lives at [elastic.co/docs/api/](https://www.elastic.co/docs/api/) and is hosted by Bump.sh.
+
+## Example: Elasticsearch
+
+The Elasticsearch API specification workflow is particularly complex because it combines multiple source files and serves various downstream consumers. 
+
+When adding a new API, Elasticsearch engineers first create a very basic spec in the [elasticsearch repo](https://github.com/elastic/elasticsearch/tree/main/rest-api-spec). Those specs are mirrored, and made more robust and detailed in [elasticsearch-specification](https://github.com/elastic/elasticsearch-specification/tree/main/docs).
+
+The generated Schema JSON and OpenAPI documents feed into client libraries (and their docs), the Dev Tools Console, and the [Elasticsearch API reference](https://www.elastic.co/docs/api/doc/elasticsearch/) (including the [Serverless API reference](https://www.elastic.co/docs/api/doc/elasticsearch-serverless/)). Here's how the pipeline works:
+
+% Mermaid source code
+<!-- 
+flowchart TD
+    A[TypeScript API definitions] -->|compiled| B[Schema JSON]
+    F[JSON spec files] -->|merged| B
+    G[Examples YAML files] -->|processed| B
+    B -->|converted| C1[OpenAPI for Documentation]
+    B -->|converted| C2[OpenAPI for Clients, Dev Tools Console, etc.]
+    H[Overlays] -->|applied| C1
+    C1 -->|published| E["HTML published by Bump.sh"]
+    C2 -->|used by| I[Clients & Dev Tools Console]
+    B -->|generated| D[Client Libraries]
+    
+    style A fill:#fff2cc
+    style F fill:#fff2cc
+    style G fill:#fff2cc
+    style B fill:#d5e8d4
+    style C1 fill:#e1d5e7
+    style C2 fill:#e1d5e7
+    style H fill:#fff2cc
+    style D fill:#dae8fc
+    style E fill:#dae8fc
+    style I fill:#dae8fc
+-->
+
+
+:::{image} images/es-api-docs-pipeline.png
+:alt: API docs generation pipeline diagram, showing the flow from TypeScript specifications to JSON schema, OpenAPI transformation, and HTML publishing.
+:width: 750px
+:align: center
+:::
+
+:::{important}
```

Comment:
> :::{important}

I think we might not want to commit this date here, since it's not set in stone AFAIK